### PR TITLE
[TF] Add max_instances to cloudfunctions functions

### PR DESCRIPTION
**Pull request by emilymye**
Fixes https://github.com/terraform-providers/terraform-provider-google/issues/3430

Also fix enviroment variables update mask (should be camel not snake case)

-----------------------------------------------------------------
# [all]
## [terraform]
Add max instances to cloud functions function
### [terraform-beta]
Add max instances to cloud functions function

### CODE REVIEW

File: third_party/terraform/resources/resource_cloudfunctions_function.go [hunk 1]
```diff
@@ -275,6 +275,13 @@ func resourceCloudFunctionsFunction() *schema.Resource {
 				Removed:  "This field is removed. Use `event_trigger.failure_policy.retry` instead.",
 			},
 
+			"max_instances": {
+				Type:             schema.TypeInt,
+				Optional:         true,
+				ValidateFunc:     validation.IntAtLeast(0),
+				DiffSuppressFunc: emptyOrDefaultStringSuppress("0"),
```
Comment (rileykarson) [inline third_party/terraform/resources/resource_cloudfunctions_function.go]: Isn't the default `0`? And isn't this an int, not a string?

File: third_party/terraform/resources/resource_cloudfunctions_function.go [hunk 2]
```diff
@@ -507,14 +518,19 @@ func resourceCloudFunctionsUpdate(d *schema.ResourceData, meta interface{}) erro
 
 	if d.HasChange("environment_variables") {
 		function.EnvironmentVariables = expandEnvironmentVariables(d)
-		updateMaskArr = append(updateMaskArr, "environment_variables")
+		updateMaskArr = append(updateMaskArr, "environmentVariables")
```
Comment (rileykarson) [inline third_party/terraform/resources/resource_cloudfunctions_function.go]: RIP

File: third_party/terraform/resources/resource_cloudfunctions_function.go (hunk 1 above)
Comment (emilymye) [reply third_party/terraform/resources/resource_cloudfunctions_function.go]: Technically if default is 0 it'll always send 0 vs not sending it? But they act the same and eventually it'll be 0 so it's nbd, I'll change it. It'll let people unset it that way (when they remove it). As for DSF, diffs are done on the string values from state/config (function signature is `func(k, old, new string, ...)`. Default diff does per-type comparison (but still uses diffString)

File: third_party/terraform/resources/resource_cloudfunctions_function.go (hunk 2 above)
Comment (emilymye) [reply third_party/terraform/resources/resource_cloudfunctions_function.go]: 😢

**Pull request merged**